### PR TITLE
Rewrite push_image_to_ecr.sh script

### PR DESCRIPTION
**Pull request by VSpike**
Rewrite this script, remove the flag parsing. Script now is convention-driven, taking its configuration from the usual environment variables.

### CODE REVIEW

File: scripts/common [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+if [[ -t 2 || -v FORCE_COLOUR ]]; then
```
Comment (mitchfriedman) [inline scripts/common]: colors 😍

File: scripts/push_image_to_ecr.sh [hunk 2]
```diff
@@ -1,67 +1,126 @@
 #!/usr/bin/env bash
 
-set -e
+## Script to push local docker images to ECR
+#
+# Variables:-
+#
+# * $AWS_ACCESS_KEY_ID - AWS access key (required)
```
Comment (mitchfriedman) [inline scripts/push_image_to_ecr.sh]: so just to confirm, the latest `roo_app` (and other modules) use these as the exported environment variables, is that right?

**Comment by VSpike** [comment]
> My bash isn't great, so I don't have any feedback on the actual code. From the perspective of a consumer, this looks easier to use and less likely to mess something up, as long as these dependent implicit variables are the ones that are exported by the most recent application TF templates

Yeah, I looked through the existing CircleCI configs that we had. A couple of those variables are added by TF modules, and some come from the config file itself. I found that even in the case of variables local to the CircleCI config, in 9 out of 10 cases people use the same names, meaning that in most cases this script can be used with no parameters at all.

File: scripts/push_image_to_ecr.sh (hunk 2 above)
Comment (VSpike) [reply scripts/push_image_to_ecr.sh]: Yep.

* Module `circleci_iam_user` exports:
  - `${var.namespace}_AWS_ACCESS_KEY_ID`
  - `${var.namespace}_AWS_SECRET_ACCESS_KEY`
* Module `docker_registry` exports:
  - `${var.namespace}_AWS_ECR_REPO_URL`
  - `${var.namespace}_AWS_ECR_REPO_REGION`
* Module `roo_app_basic` exports:
  - `${var.namespace}_AWS_ACCESS_KEY_ID`
  - `${var.namespace}_AWS_SECRET_ACCESS_KEY`
  - `${var.namespace}_AWS_ECR_REPO_URL`
  - `${var.namespace}_AWS_REGION`

The script will accept all of these.

Comment (mitchfriedman) [reply scripts/push_image_to_ecr.sh]: This is perfect then 👍

**Pull request merged**